petab/visualize/plotting_config.py

```diff
@@ -23,158 +19,138 @@ def plot_lowlevel(vis_spec: pd.DataFrame,
     Parameters
     ----------
 
-    vis_spec:
-        pandas data frame, contains defined data format (visualization file)
+    plot_spec:
+        contains defined data format (visualization file)
     ax:
-        np.ndarray, matplotlib.Axes
-    axx:
-        int, subplot axis indices for x
-    axy:
-        int, subplot axis indices for y
+        axes to which to plot
     conditions:
         pd.Series, Values on x-axis
     ms:
         pd.DataFrame,  containing measurement data which should be plotted
-    ind_plot:
-        pd.Series, boolean vector, with size:
-        len(rows in visualization file) x 1
-        with 'True' entries for rows which should be plotted
-    i_visu_spec:
-        int64, current index (row number) of row which should be plotted in
-        visualizationSpecification file
     plot_sim:
         bool, tells whether or not simulated data should be plotted as well
-
-    Returns
-    -------
-    ax: matplotlib.Axes
     """
 
-    # set xScale
-    if vis_spec[X_SCALE][i_visu_spec] == LIN:
-        ax[axx, axy].set_xscale("linear")
-    elif vis_spec[X_SCALE][i_visu_spec] == LOG10:
-        ax[axx, axy].set_xscale("log")
-    elif vis_spec.xScale[i_visu_spec] == LOG:
-        ax[axx, axy].set_xscale("log", basex=np.e)
-
     # set yScale
-    if vis_spec.yScale[i_visu_spec] == LIN:
-        ax[axx, axy].set_yscale("linear")
-    elif vis_spec.yScale[i_visu_spec] == LOG10:
-        ax[axx, axy].set_yscale("log")
-    elif vis_spec.yScale[i_visu_spec] == LOG:
-        ax[axx, axy].set_yscale("log", basey=np.e)
+    if plot_spec.yScale == 'lin':
+        ax.set_yscale("linear")
+    elif plot_spec.yScale == 'log10':
+        ax.set_yscale("log")
+    elif plot_spec.yScale == 'log':
+        ax.set_yscale("log", basey=np.e)
 
```

Comment:
Here something is going wrong in the plotting to file case:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-11b209bfca12> in <module>
     11                               visualization_file_path,
     12                               simulation_file_path,
---> 13                               subplot_file_path="./"
     14                               )
     15 plt.show()

~/git/PEtab-FF/petab/visualize/plot_data_and_simulation.py in plot_data_and_simulation(exp_data, exp_conditions, vis_spec, sim_data, dataset_id_list, sim_cond_id_list, sim_cond_num_list, observable_id_list, observable_num_list, plotted_noise, subplot_file_path)
    142             # handle plot of current dataset
    143             handle_dataset_plot(plot_spec, ax, exp_data,
--> 144                                 exp_conditions, sim_data)
    145 
    146         if BAR_PLOT in vis_spec[ind_plot][PLOT_TYPE_SIMULATION].values:

~/git/PEtab-FF/petab/visualize/helper_functions.py in handle_dataset_plot(plot_spec, ax, exp_data, exp_conditions, sim_data)
    381     nan_set = all([np.isnan(val) for val in measurement_to_plot['mean']])
    382     if not nan_set:
--> 383         plot_lowlevel(plot_spec, ax, conditions, measurement_to_plot, plot_sim)
    384 
    385     # Beautify plots

~/git/PEtab-FF/petab/visualize/plotting_config.py in plot_lowlevel(plot_spec, ax, conditions, ms, plot_sim)
     34     # set yScale
     35     if plot_spec.yScale == 'lin':
---> 36         ax.set_yscale("linear")
     37     elif plot_spec.yScale == 'log10':
     38         ax.set_yscale("log")

AttributeError: 'numpy.ndarray' object has no attribute 'set_yscale'

```

Reply:
what testcase is that? I ran all tests from both visualization tests and everything passed for me

Reply:
The Iseense example (notebook) with option save to file switched on (sub_plot_file != "")
```
from petab.visualize import plot_data_and_simulation
import matplotlib.pyplot as plt
folder = "example_Isensee/"

data_file_path = folder + "Isensee_measurementData.tsv"
condition_file_path = folder + "Isensee_experimentalCondition.tsv"
visualization_file_path = folder + "Isensee_visualizationSpecification.tsv"
simulation_file_path = folder + "Isensee_simulationData.tsv"

# function to call, to plot data and simulations
ax = plot_data_and_simulation(data_file_path,
                              condition_file_path,
                              visualization_file_path,
                              simulation_file_path,
                              subplot_file_path="./")
```

Reply:
sounds like a good testcase to add

Reply:
fixed in most recent commit

Reply:
thanks, looks good now